app/config/config.yml

```diff
@@ -66,3 +66,15 @@ swiftmailer:
     username:  "%mailer_user%"
     password:  "%mailer_password%"
     spool:     { type: memory }
+
+# Can be changed directly in the bundle is we found an agreement
+dunglas_action:
+    autodiscover:
+        enabled:   true
+        directories:
+            action: [ ]
+            console: [ Command ]
+    directories:         # List of directories relative to the kernel root directory containing classes to auto-register.
+        action: [ '../src/AppBundle/Controller' ] # Still an autowiring_type definition to add in Symfony to avoid this
```

Comment:
I didn't expect the configuration to work like this. It had never occurred to me to do any sort of autowiring across *all* bundles - I initially don't like that idea, and it complicates the configuration (I had to research what the difference was between the `autodiscover` and `directories` keys). 

Also:

- what causes console commands to autowire? Are the `action` and `console` keys significant? I think they are. If so, it's weird to me to *also* allow other keys - like `foo` (used in the bundle README). I understand that this is to allow *other* directories to be autowired, but I don't like that *sometimes* the keys are special, and sometimes not. It also means we can't catch typos.

- You mentioned `Still an autowiring_type definition to add in Symfony to avoid this` - what
does this refer to?

- Would it be possible to avoid the special `action-annotation` route type altogether? In other words, extend `@Route` to use a service definition for the controller if one exists? I ask because it *feels* like there are two ways to auto-register a controller as a service: either by using some config in `config.yml` *or* by using the `action-annotation`. The line between what each does is blurry. I think extending `@Route` may be a non-trivial task, but it makes a lot more sense to me: it's in the spirit of autowiring where we (in this case the `@Route` system) effectively "asks" the container for the service for an instance and it returns it.